backend/proxy/secure_socks_proxy.go

```diff
@@ -37,62 +36,65 @@ var (
 	PluginSecureSocksProxyServerName = "GF_SECURE_SOCKS_DATASOURCE_PROXY_SERVER_NAME"
 )
 
-// SecureSocksProxyConfig contains the information needed to allow datasource connections to be
-// proxied to a secure socks proxy
-type secureSocksProxyConfig struct {
-	clientCert   string
-	clientKey    string
-	rootCA       string
-	proxyAddress string
-	serverName   string
+// Client is the main Proxy Client interface.
+type Client interface {
+	SecureSocksProxyEnabled(opts *Options) bool
+	ConfigureSecureSocksHTTPProxy(transport *http.Transport, opts *Options) error
+	NewSecureSocksProxyContextDialer(opts *Options) (proxy.Dialer, error)
 }
 
-// SecureSocksProxyEnabled checks if the Grafana instance allows the secure socks proxy to be used
-// and the datasource options specify to use the proxy
-func SecureSocksProxyEnabled(opts *Options) bool {
-	if opts == nil {
-		return false
-	}
+// ClientCfg contains the information needed to allow datasource connections to be
+// proxied to a secure socks proxy.
+type ClientCfg struct {
+	Enabled      bool
+	ClientCert   string
+	ClientKey    string
+	RootCA       string
+	ProxyAddress string
+	ServerName   string
+}
 
-	if !opts.Enabled {
-		return false
-	}
+// Cli is the default Proxy Client.
+var Cli = New()
 
-	if value, ok := os.LookupEnv(PluginSecureSocksProxyEnabled); ok {
-		enabledOnInst, err := strconv.ParseBool(value)
-		if err != nil {
-			return false
-		}
+// New creates a new proxy client from the environment variables set by the grafana-server in the plugin.
+func New() Client {
+	return NewWithCfg(getConfigFromEnv())
+}
 
-		return enabledOnInst
+// NewWithCfg creates a new proxy client from a given config.
+func NewWithCfg(cfg *ClientCfg) Client {
+	return &cfgProxyWrapper{
+		cfg: cfg,
 	}
+}
 
-	return false
+type cfgProxyWrapper struct {
+	cfg *ClientCfg
 }
 
-// SecureSocksProxyEnabledOnDS checks the datasource json data for `enableSecureSocksProxy`
-// to determine if the secure socks proxy should be enabled on it
-func SecureSocksProxyEnabledOnDS(jsonData map[string]interface{}) bool {
-	res, enabled := jsonData["enableSecureSocksProxy"]
-	if !enabled {
+// SecureSocksProxyEnabled checks if the Grafana instance allows the secure socks proxy to be used
+// and the datasource options specify to use the proxy
+func (p *cfgProxyWrapper) SecureSocksProxyEnabled(opts *Options) bool {
+	if opts == nil {
 		return false
 	}
 
-	if val, ok := res.(bool); ok {
-		return val
+	if !opts.Enabled {
+		return false
 	}
 
-	return false
+	return (p.cfg != nil) && p.cfg.Enabled
```

Comment:
The proxy has to be enabled / set up on Grafana itself, and then the datasource has to opt to use that proxy. But the order is confusing here, because it has to be enabled on Grafana to be enabled on the datasource. I changed the ordering & added comments. Thanks for calling this out!